map/bookmark_helpers.cpp

```diff
@@ -245,17 +245,23 @@ std::string RemoveInvalidSymbols(std::string const & name)
   return strings::ToUtf8(filtered);
 }
 
-std::string GenerateUniqueFileName(const std::string & path, std::string name, std::string const & ext)
+std::string BuildName(std::string name, std::string const & suffix, std::string_view const ext)
+{
+  return name.append(suffix).append(ext);
+}
+
+std::string GenerateUniqueFileName(std::string const & path, std::string name, std::string_view const ext)
 {
   // Remove extension, if file name already contains it.
   if (strings::EndsWith(name, ext))
     name.resize(name.size() - ext.size());
 
   size_t counter = 1;
   std::string suffix;
-  while (Platform::IsFileExistsByFullPath(base::JoinPath(path, name + suffix + ext)))
+
+  while (Platform::IsFileExistsByFullPath(base::JoinPath(path, BuildName(name, suffix, ext))))
     suffix = strings::to_string(counter++);
-  return base::JoinPath(path, name + suffix + ext);
+  return base::JoinPath(path, BuildName(name, suffix, ext));
```

Comment:
I'm not insisting, I assumed that it can be easy, but looks like the reality is worse )

Anyway, if we are already here, it makes sense to fix it properly. I hope that @cyber-toad learned something new in the process )

Reply:
So take my snippet to avoid twice calculation of the same path?

Reply:
Done